v2/internal/parse_test.go

```diff
@@ -28,23 +33,182 @@ func TestParseMessageFileBytes(t *testing.T) {
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:          "notnested",
+					Description: "world",
+				}},
+			},
+		},
+		{
+			name: "basic test with id",
+			file: `{"key": {"id": "forced.id"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID: "forced.id",
+				}},
+			},
+		},
+		{
+			name: "basic test with description and dummy",
+			file: `{"notnested": {"description": "world", "dummy": "nothing"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:          "notnested",
+					Description: "world",
+				}},
+			},
+		},
+		{
+			name: "deeply nested test",
+			file: `{"outer": {"nested": {"inner": "value"}}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
+		{
+			name: "multiple nested test",
+			file: `{"nested": {"hello": "world", "bye": "all"}}`,
+			path: "en.json",
+			messageFile: &MessageFile{
+				Path:   "en.json",
+				Tag:    language.English,
+				Format: "json",
+				Messages: []*Message{{
+					ID:    "nested.hello",
+					Other: "world",
+				}, {
+					ID:    "nested.bye",
+					Other: "all",
+				}},
+			},
+		},
+		{
+			name: "YAML nested test",
+			file: `
+outer:
+    nested:
+        inner: "value"`,
+			path:           "en.yaml",
+			unmarshalFuncs: map[string]UnmarshalFunc{"yaml": yaml.Unmarshal},
+			messageFile: &MessageFile{
+				Path:   "en.yaml",
+				Tag:    language.English,
+				Format: "yaml",
+				Messages: []*Message{{
+					ID:    "outer.nested.inner",
+					Other: "value",
+				}},
+			},
+		},
 	}
 	for _, testCase := range testCases {
-		actual, err := ParseMessageFileBytes([]byte(testCase.file), testCase.path, testCase.unmarshalFuncs)
-		if err != testCase.err {
-			t.Fatalf("expected error %#v; got %#v", testCase.err, err)
-		}
-		if actual.Path != testCase.messageFile.Path {
-			t.Fatalf("expected path %q; got %q", testCase.messageFile.Path, actual.Path)
-		}
-		if actual.Tag != testCase.messageFile.Tag {
-			t.Fatalf("expected tag %q; got %q", testCase.messageFile.Tag, actual.Tag)
-		}
-		if actual.Format != testCase.messageFile.Format {
-			t.Fatalf("expected format %q; got %q", testCase.messageFile.Format, actual.Format)
+		t.Run(testCase.name, func(t *testing.T) {
+			actual, err := ParseMessageFileBytes([]byte(testCase.file), testCase.path, testCase.unmarshalFuncs)
+			if (err == nil && testCase.err != nil) ||
+				(err != nil && testCase.err == nil) ||
+				(err != nil && testCase.err != nil && err.Error() != testCase.err.Error()) {
+				t.Errorf("expected error %#v; got %#v", testCase.err, err)
```

Comment:
Might as well use `t.Fatalf` and omit the return.

Reply:
Indeed, I also changed that in commit beea385.

I kept the last outputs as `Errorf()` to allow to continue with next assertion: only the error and messageFile tests are fatal and prevent to continue.